include/psim/core/simulation.hpp

```diff
@@ -32,54 +32,53 @@
 #include <psim/core/model.hpp>
 #include <psim/core/state.hpp>
 
-#include <memory>
-#include <utility>
-
 namespace psim {
 
 /** @brief Represents a full simulation.
+ *
+ *  @tparam C Underlying model used by the simulation.
  *
  *  The main purpose of the simulation class is to merge the functionality of a
  *  state and model. It's how end users are intended to interact with models,
  *  state fields, and steps.
  */
+template <class C>
 class Simulation : public State {
  private:
-  /** @brief Model employed by the simulation.
-   *
-   *  Making this field dynamically allocated was very intentional. It ensures
-   *  on moves that the stored state fields address remain valid.
+  /** @brief Random number generator shared across the entire simulation.
    */
-  std::unique_ptr<Model> _model;
+  RandomsGenerator _randoms;
 
- protected:
-  /** @brief Private constructor employed by the factory function.
+  /** @brief Model employed by the simulation.
    */
-  Simulation(std::unique_ptr<Model> &&model);
+  C _model;
 
  public:
   Simulation() = delete;
   Simulation(Simulation const &) = delete;
-  Simulation(Simulation &&) = default;
+  Simulation(Simulation &&) = delete;
   Simulation &operator=(Simulation const &) = delete;
-  Simulation &operator=(Simulation &&) = default;
+  Simulation &operator=(Simulation &&) = delete;
```

Comment:
These are deleted now so that the references to the randoms generator in the model isn't invalidated by a move or copy.